### DISCUSSION TITLE
GDK/GameCore support for Gaming.Desktop.x64

### DISCUSSION THREAD

**Discussion by DougGal-Microsoft** [microsoft/msquic#1972] [General]
In addition to compiling against the WINAPI_GAMES partition for GameCore/GDK, it turns out that we also need to link against the GDK umbrella version of the Windows SDK link libraries in addition to restricting slightly the instruction set that the compiler may use to ensure that the resulting binary will be able to run on consoles.  The public version of the GDK (https://github.com/microsoft/GDK) provides the Gaming.Desktop.x64 build target in Visual Studio to achieve this.  What is the correct way to add this build target to the msquic GameCore build?

There also are Gaming.Xbox.Scarlett.x64 and Gaming.Xbox.XboxOne.x64 build targets in the version of the GDK gated behind the Xbox developer preview that provide certain relatively uncommon vector instruction extensions and other performance improvements for binaries targeted specifically at the consoles.  This is not necessary (the Gaming.Desktop.x64 build target is a subset that works everywhere), but could be a performance improvement if msquic has a way of taking advantage of these math operations.  It does, however, come with a"not public" requirement.

**Comment by thhous-msft** [comment]
https://github.com/microsoft/Xbox-GDK-Samples/tree/main/Samples/Tools/CMakeExample

Here's the CMake sample to get the right flags.
